File: go/basicResponse.go
package utils

type BasicResponse struct {
	Success bool        `json:"success"`
	Message *string     `json:"message"`
	Data    interface{} `json:"data"`
}

func GetBasicResponse(response BasicResponse) BasicResponse {
	return BasicResponse{
		Success: response.Success,
		Message: response.Message,
		Data:    response.Data,
	}
}

File: java/BasicResponse.java
// compileOnly 'org.projectlombok:lombok'
// annotationProcessor 'org.projectlombok:lombok'
// com.google.gson.Gson

import com.google.gson.Gson;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Optional;

@Data
@Getter
@Setter
@Builder
public class BasicResponse {
    private boolean success;
    private Optional<String> message;
    private Optional<Object> data;

    public String toJson() {
        Gson gson = new Gson();

        return gson.toJson(this);
    }
}
